src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Documents/VisualStudioEditorDocumentManager.cs

```diff
@@ -63,6 +61,8 @@ internal class VisualStudioEditorDocumentManager : EditorDocumentManagerBase
 
         protected override ITextBuffer GetTextBufferForOpenDocument(string filePath)
         {
+            JoinableTaskContext.AssertUIThread();
```

Comment:
Maybe we should add comments above each call to AssertUIThread with the reason/call that it's needed for? I would GUESS it's the calls to `_runningDocumentTable` in this instance, but I'm not sure.